File: plugins/dev-standards/evals/developer-csharp/fixtures/OrderService.cs
using Microsoft.Extensions.Logging;

namespace Shop.Orders;

public class OrderService
{
    private readonly IOrderRepository _repository;
    private readonly IPaymentGateway _gateway;
    private readonly ILogger<OrderService> _logger;

    public OrderService(IOrderRepository repository, IPaymentGateway gateway, ILogger<OrderService> logger)
    {
        _repository = repository;
        _gateway = gateway;
        _logger = logger;
    }

    // CancellationToken accepted then dropped: not passed to any awaited call below.
    public async Task<Order> PlaceAsync(OrderRequest request, CancellationToken cancellationToken)
    {
        var customer = _repository.FindCustomer(request.CustomerId).Result;

        _logger.LogInformation($"Placing order for customer {customer.Email} total {request.Total}");

        var charge = _gateway.ChargeAsync(request.Total).GetAwaiter().GetResult();
        if (!charge.Succeeded)
        {
            throw new Exception("payment failed");
        }

        var order = new Order(Guid.NewGuid(), request.CustomerId, request.Total);
        _repository.SaveAsync(order).Wait();

        NotifyWarehouse(order);
        return await Task.FromResult(order);
    }

    // async void outside an event handler: exceptions here cannot be observed by the caller.
    private async void NotifyWarehouse(Order order)
    {
        try
        {
            await _gateway.NotifyAsync(order.Id);
        }
        catch (Exception)
        {
            // swallowed
        }
    }
}

File: plugins/dev-standards/evals/developer-python/fixtures/order_service.py
import logging

logger = logging.getLogger(__name__)


class OrderService:
    def __init__(self, repository, gateway):
        self.repository = repository
        self.gateway = gateway

    # no type hints, no return annotation, payload is a raw dict
    def place(self, payload):
        total = payload["total"]
        email = payload["customer"]["email"]
        card = payload["customer"]["card_number"]

        logger.info("placing order for %s card %s total %s", email, card, total)

        if total > 1000:
            payload["discount"] = total * 0.05

        try:
            charge = self.gateway.charge(total)
        except Exception:
            return None

        if not charge["ok"]:
            raise Exception("payment failed")

        order = self.repository.save(payload)
        return {"id": order["id"], "total": total}

    def cancel(self, order_id):
        try:
            self.repository.delete(order_id)
        except Exception as exc:
            logger.debug(exc)
        return True
